text/0000-ppv.md

```diff
@@ -0,0 +1,1596 @@
+///
+/// # Precondition
+///
+/// If `slice.len() < Self::lanes()` the behavior is undefined.
+pub unsafe fn write_unaligned_unchecked(self, slice: &mut [element_type]);
+
+/// Instantiates a new vector with the values of the `slice` without 
+/// moving them, leaving the memory in `slice` unchanged.
+///
+/// # Panics
+///
+/// If `slice.len() != Self::lanes()` or `&slice[0]` is not aligned
+/// to an `align_of::<Self>()` boundary.
+pub fn read_aligned(slice: &[element_type]) -> Self;
+
+/// Instantiates a new vector with the values of the `slice` without 
+/// moving them, leaving the memory in `slice` unchanged.
+///
+/// # Panics
+///
+/// If `slice.len() != Self::lanes()`.
+pub fn read_unaligned(slice: &[element_type]) -> Self;
+
+/// Instantiates a new vector with the values of the `slice` without 
+/// moving them, leaving the memory in `slice` unchanged.
+///
+/// # Precondition
+///
+/// If `slice.len() < Self::lanes()` or `&slice[0]` is not aligned
+/// to an `align_of::<Self>()` boundary, the behavior is undefined.
+pub unsafe fn read_aligned_unchecked(slice: &[element_type]) -> Self;
+
+/// Instantiates a new vector with the values of the `slice` without 
+/// moving them, leaving the memory in `slice` unchanged.
+///
+/// # Precondition
+///
+/// If `slice.len() < Self::lanes()` the behavior is undefined.
+pub unsafe fn read_unaligned_unchecked(slice: &[element_type]) -> Self;
+}
+```
+
+##### Discontinuous masked reads and writes (scatter and gather)
+
+Vector masks implement the following methods:
+
+```rust
+impl m{lane_width}x{number_of_lanes} {
+/// Instantiates a new vector with the values of the `slice` located at 
+/// the `offset`s without moving them for which the mask (`self`) is `true`
+/// and with the values of `default` otherwise. The memory of the `slice` at 
+/// the `offsets` for which the mask is `false` is not read.
+///
+/// # Precondition
+///
+/// If `slice.len() < offset.max_element()` the behavior is undefined.
+pub unsafe fn read_scattered_unchecked<T, O, D>(self, slice: &[T], offset: O, default: D) -> D
+    where <implementation defined> 
+        // for exposition only:
+        // number_of_lanes == D::lanes() == O::lanes(), 
+        // D::element_type == T,
+        // O::element_type == usize,
+;
+
+/// Writes the elements of the vector `values` for which the mask (`self`) 
+/// is `true` to the `slice` at `offset`s without reading or dropping 
+/// the old values. No memory is written to the `slice` elements at 
+/// the `offset`s for which the mask is `false`.
+///
+/// If multiple `offset`s have the same value, that is, if multiple lanes 
+/// from `values` are to be written to the same memory location, the writes
+/// are ordered from least significant to most significant element.
+///
+/// # Precondition
+///
+/// If `slice.len() < offset.max_element()` the behavior is undefined.
+pub unsafe fn write_scattered_unchecked<T, O, D>(self, slice: &mut [T], offset: O, values: D)
```

Comment:
Same interface concern as with gather, and in addition (as with the other write_unchecked methods) this interface can't soundly be used to initialize memory that was previously uninitialized.